From d0528fb5e51f0e69af289f19cae434e43c10584f Mon Sep 17 00:00:00 2001
From: Vinay Gannevaram <quic_vganneva@quicinc.com>
Date: Sun, 19 Jan 2025 13:18:46 +0530
Subject: [PATCH] Add QCA vendor feature flags to indicate P2P-R2 and PCC
 support

Add feature flags to indicate support for P2P-R2 and P2P
Connection Compatibility feature.

Signed-off-by: Vinay Gannevaram <quic_vganneva@quicinc.com>
---
 src/common/qca-vendor.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/common/qca-vendor.h b/src/common/qca-vendor.h
index 99ba6becc..3cc2f93c6 100644
--- a/src/common/qca-vendor.h
+++ b/src/common/qca-vendor.h
@@ -2343,6 +2343,13 @@ enum qca_wlan_vendor_acs_hw_mode {
  *	supports preferring 6 GHz PSC channel as a primary channel in ACS
  *	result.
  *
+ * @QCA_WLAN_VENDOR_FEATURE_P2P_V2: Flag indicates that the driver supports
+ *	P2P R2 functionality (P2P R2 Discovery, Pairing, TWT power save, etc).
+ *
+ * @QCA_WLAN_VENDOR_FEATURE_PCC_MODE: Flag indicates that the driver supports
+ *	P2P Connection Compatibility mode in which GO allows connection
+ *	with both P2P R1 and R2 clients.
+ *
  * @NUM_QCA_WLAN_VENDOR_FEATURES: Number of assigned feature bits
  */
 enum qca_wlan_vendor_features {
@@ -2374,6 +2381,8 @@ enum qca_wlan_vendor_features {
 	QCA_WLAN_VENDOR_FEATURE_RSN_OVERRIDE_STA = 25,
 	QCA_WLAN_VENDOR_FEATURE_NAN_USD_OFFLOAD = 26,
 	QCA_WLAN_VENDOR_FEATURE_ACS_PREFER_6GHZ_PSC = 27,
+	QCA_WLAN_VENDOR_FEATURE_P2P_V2 = 28,
+	QCA_WLAN_VENDOR_FEATURE_PCC_MODE = 29,
 	NUM_QCA_WLAN_VENDOR_FEATURES /* keep last */
 };
 
-- 
2.43.0

